mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -240,28 +243,37 @@ async fn block_header_utxo_loop<T: UtxoCommonOps>(
             Err(e) => {
                 error!("Error {} on getting the height of the last stored header in DB!", e);
                 sync_status_loop_handle.notify_on_temp_error(e.to_string());
-                Timer::sleep(10.).await;
+                Timer::sleep(BLOCK_HEADERS_LOOP_ERORR_SLEEP_TIMER).await;
                 continue;
             },
         };
 
-        // Todo: Add code for the case if a chain reorganization happens
-        if from_block_height == last_block_height {
-            sync_status_loop_handle.notify_sync_finished(last_block_height);
-            last_block_height = match coin.as_ref().rpc_client.get_block_count().compat().await {
+        let mut to_block_height = from_block_height + chunk_size;
+        if to_block_height > block_count {
```

Comment:
It seems I forgot to use `drop_mutability!` macro. Could you please use it after this block?